### PR TITLE
Add Newsletter block

### PR DESCRIPTION
**Pull request by saugatmalla**
Refer to @rgclayton & @auniverseaway for newsletter subscribe API testing. 

Test URLs: 
- Before: https://main--blog--adobecom.hlx.page/drafts/smalla/newsletter-modal
- After: https://newsletter--blog--adobecom.hlx.page/drafts/smalla/newsletter-modal

### CODE REVIEW

File: blocks/newsletter-modal/newsletter-modal.css [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+  overflow: hidden;
+}
+
+.newsletter-modal-banner-container {
+  width: 100%;
+}
+
+.newsletter-modal-banner-container img {
+  width: 100%;
+}
```
Comment (meganthecoder) [inline blocks/newsletter-modal/newsletter-modal.css:19]: Would you want to consolidate these?
```suggestion
.newsletter-modal-banner-container,
.newsletter-modal-banner-container img {
  width: 100%;
}
```

File: blocks/newsletter-modal/newsletter-modal.css [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+  align-items: center;
+  justify-content: center;
+  min-height: 365px;
+}
+
+.newsletter-modal-text {
+  width: 80%;
+  max-width: 420px;
+  font-size: var(--type-body-l-size);
+  margin: 0 0 var(--spacing-xs) !important;
```
Comment (meganthecoder) [inline blocks/newsletter-modal/newsletter-modal.css:34]: Could you avoid `!important` by using more selector specification?

File: blocks/newsletter-modal/newsletter-modal.css [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+.newsletter-modal .newsletter-modal-cta {
+  flex-shrink: 0;
+  background-color: var(--link-color-dark);
+  color: var(--color-white);
+  border-radius: 100px;
+  padding: var(--spacing-xxs) var(--spacing-xs);
+  border: none;
+  cursor: pointer;
+  margin: 0 0 15px 15px;
+  font-family: var(--body-font-family);
```
Comment (meganthecoder) [inline blocks/newsletter-modal/newsletter-modal.css:91]: Seems strange that you have to keep setting the font family

File: blocks/newsletter-modal/newsletter-modal.css [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+.newsletter-modal-confirmation-close:hover {
+  background-color: var(--link-color-dark) ;
+  color: var(--color-white);
+}
+
+@media (min-width: 600px) {
+  .newsletter-modal {
+      max-width: 440px;
+      border-radius: 6px;
+      height: auto!important;
```
Comment (meganthecoder) [inline blocks/newsletter-modal/newsletter-modal.css:140]: Again, should avoid important if at all possible

File: blocks/newsletter-modal/newsletter-modal.js [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+
+  const emailText = children[2].querySelector('div p:nth-child(1)').textContent;
+  const emailPlaceholder = children[2].querySelector('div p:nth-child(2)').textContent;
+  const cta = children[2].querySelector('div p:nth-child(3)').textContent;
+
+  const successMsg = children[3].querySelector(':scope > div:nth-child(1) ').textContent;
+  const errorMsg = children[3].querySelector(':scope > div:nth-child(2) ').textContent;
+
+  const bannerContainer = createTag('div', { class: 'newsletter-modal-banner-container' }, picture);
+  const content = createTag('div', { class: 'newsletter-modal-content' });
```
Comment (meganthecoder) [inline blocks/newsletter-modal/newsletter-modal.js:48]: It seems somewhat wasteful to remove divs and then create new ones. On the other hand, it's potentially easier for devs to follow what's going on that way. I'd be curious to hear what other milo devs prefer.

File: blocks/newsletter-modal/newsletter-modal.js [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+          displayConfirmation(content, successMsg, createTag);
+        })
+        .catch(() => {
+          displayConfirmation(content, errorMsg, createTag);
+        });
+    } else {
+      emailEl.classList.add('error');
+      emailEl.reportValidity();
+    }
+  });
```
Comment (meganthecoder) [inline blocks/newsletter-modal/newsletter-modal.js:94]: I think the listener's function should be pulled out. It adds a lot of bulk to `init`

File: blocks/newsletter-modal/newsletter-modal.js [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+    const email = emailEl.value;
+
+    if (email && emailEl.checkValidity()) {
+      const headers = new Headers();
+      headers.append('Content-Type', 'application/json');
+
+      const body = {
+        sname: 'adbeblog',
+        email,
+        consent_notice: '<div class="disclaimer detail-spectrum-m" style="letter-spacing: 0px; padding-top: 15px;">The Adobe family of companies may keep me informed with personalized emails from the Adobe Blog team. See our <a href="https://www.adobe.com/privacy/policy.html" target="_blank">Privacy Policy</a> for more details or to opt-out at any time.</div>',
```
Comment (meganthecoder) [inline blocks/newsletter-modal/newsletter-modal.js:73]: Do we need to worry about localization?

File: blocks/newsletter-modal/newsletter-modal.js [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+  if (qModals?.length) {
+    qModals.forEach((modal) => {
+      if (modal.nextElementSibling?.classList.contains('modal-curtain')) {
+        modal.nextElementSibling.remove();
+      }
+      modal.remove();
+    });
+    window.history.pushState('', document.title, `${window.location.pathname}${window.location.search}`);
+  }
+}
```
Comment (meganthecoder) [inline blocks/newsletter-modal/newsletter-modal.js:14]: Why recreate the closeModal?

File: blocks/newsletter-modal/newsletter-modal.js (hunk 7 above)
Comment (saugatmalla) [reply blocks/newsletter-modal/newsletter-modal.js:73]: Afaict, it is not required. 
@auniverseaway @rgclayton @chrischrischris Do you know about this?

File: blocks/newsletter-modal/newsletter-modal.js (hunk 8 above)
Comment (saugatmalla) [reply blocks/newsletter-modal/newsletter-modal.js:14]: This is used for the close button after the signup. Is there already a functionality for the use case?
Screen Shot 2023-03-28 at 10 44 53 AM

Comment (meganthecoder) [reply blocks/newsletter-modal/newsletter-modal.js:14]: Also, it's strange to have two close buttons but if that's that the designs say

File: blocks/newsletter-modal/newsletter-modal.js (hunk 5 above)
Comment (rgclayton) [reply blocks/newsletter-modal/newsletter-modal.js]: I tend to agree. 

If you can use what `plain.html` gives and add classes to the provided `divs`, to style correctly, that would be ideal and save some bytes.

File: blocks/newsletter-modal/newsletter-modal.js (hunk 7 above)
Comment (saugatmalla) [reply blocks/newsletter-modal/newsletter-modal.js:28]: @rgclayton Should we be concerned that the author can send anything to the API server?

Comment (rgclayton) [reply blocks/newsletter-modal/newsletter-modal.js:28]: > @rgclayton Should we be concerned that the author can send anything to the API server?

Not sure what you mean. Let's sync up.

**Review by meganthecoder — APPROVED** [review]

**Review by chrischrischris — APPROVED** [review]

**Pull request merged**